syntax = "proto3";

message Voting {

  enum VoteType{
    ACTIVE = 0;
    REJECT = 1;
  }
  enum VoteResult{
    WIN = 0;
    LOSE = 1;
    UNKNOWN = 2;
  }
  message Vote {
    string issuer_public_key = 1;
    bool accept = 2;
    uint64 timestamp = 3;
    string transaction_id = 4;
  }

  string publisher_public_key = 1;
  string elector_public_key = 2;
  VoteType vote_type = 3;
  repeated  Vote vote = 4;
  VoteResult vote_result = 5;
  uint64 close_vote_timestamp = 6;
  uint64 timestamp = 7;
  string transaction_id = 8;
}


message VotingContainer {
  repeated Voting entries = 1;
}